File: frontend/lib/models/threat_log.dart
class ThreatLog {
  final String id;
  final String timestamp;
  final String moduleType;
  final String riskLevel;
  final Map<String, dynamic> detailsJson;

  ThreatLog({
    required this.id,
    required this.timestamp,
    required this.moduleType,
    required this.riskLevel,
    required this.detailsJson,
  });

  factory ThreatLog.fromJson(Map<String, dynamic> json) {
    return ThreatLog(
      id: json['id'] as String? ?? 'Unknown',
      timestamp: json['timestamp'] as String? ?? 'Unknown',
      moduleType: json['module_type'] as String? ?? 'Unknown',
      riskLevel: json['risk_level'] as String? ?? 'Unknown',
      detailsJson: json['details_json'] as Map<String, dynamic>? ?? {},
    );
  }
}

File: frontend/app/src/main/java/com/aegis/scamfirewall/models/ThreatLog.kt
package com.aegis.scamfirewall.models

import com.google.gson.annotations.SerializedName

data class ThreatLog(
    @SerializedName("id") val id: String,
    @SerializedName("timestamp") val timestamp: String,
    @SerializedName("module_type") val moduleType: String,
    @SerializedName("risk_level") val riskLevel: String,
    @SerializedName("details_json") val detailsJson: Map<String, Any>
)
